app/src/main/java/com/example/xpvehicles/activities/RentingRequestsActivity.java

```diff
@@ -0,0 +1,163 @@
+                tvVehicleImagePosition.setText(currentPosition + " / " + totalNumberOfImages);
+            }
+        });
+    }
+
+    private void bind() {
+        topAppBarRentingRequest = findViewById(R.id.topAppBarRentingRequest);
+        tvRentingRequestVehicleName = findViewById(R.id.tvRentingRequestVehicleName);
+        tvRentingRequestVehicleDescription = findViewById(R.id.tvRentingRequestVehicleDescription);
+        tvRentingRequestDailyPrice = findViewById(R.id.tvRentingRequestDailyPrice);
+        tvOrderSummaryDailyPrice =  findViewById(R.id.tvOrderSummaryDailyPrice);
+        tvOrderSummaryNumberOfDays =  findViewById(R.id.tvOrderSummaryNumberOfDays);
+        tvOrderSummaryOrderTotal =  findViewById(R.id.tvOrderSummaryOrderTotal);
+        tvRentingRequestsPickUpDate =  findViewById(R.id.tvRentingRequestsPickUpDate);
+        tvRentingRequestReturnDate =  findViewById(R.id.tvRentingRequestReturnDate);
+        tvRentingRequestStatus =  findViewById(R.id.tvRentingRequestStatus);
+    }
+
+    private void setValues() throws ParseException {
+        // vehicle name
+        Vehicle originalVehicle = (Vehicle) rentVehicle.getVehicle();
+        String vehicleName = originalVehicle.fetchIfNeeded().getString("name");
+        tvRentingRequestVehicleName.setText(vehicleName);
+
+        // daily price
+        Number dailyPrice = originalVehicle.fetchIfNeeded().getNumber("dailyPrice");
+        tvRentingRequestDailyPrice.setText("$" + dailyPrice);
+        tvOrderSummaryDailyPrice.setText("$" + dailyPrice);
+
+        // vehicle description
+        String vehicleDescription = originalVehicle.fetchIfNeeded().getString("description");
+        tvRentingRequestVehicleDescription.setText(vehicleDescription);
+
+        SimpleDateFormat sdf = new SimpleDateFormat("MMMM dd yyyy");
+        // vehicle pickup date
+        Date pickUpDate = rentVehicle.getPickUpDate();
+        String formattedPickUpDate = sdf.format(pickUpDate);
+        tvRentingRequestsPickUpDate.setText(formattedPickUpDate);
+        // vehicle return date
+        Date returnDate = rentVehicle.getReturnDate();
+        String formattedReturnDate = sdf.format(returnDate);
+        tvRentingRequestReturnDate.setText(formattedReturnDate);
+
+        //set order summary values
+        calculateNumberOfDays(pickUpDate, returnDate);
+
+        // status
+        String status = rentVehicle.getStatus();
+        RentingStatus rentingStatus = RentingStatus.valueOf(status);
+        tvRentingRequestStatus.setText(rentingStatus.toString());
+        setStatusColor(rentingStatus);
+    }
+
+    private void calculateNumberOfDays(Date pickUpDate, Date returnDate) {
+        if (pickUpDate != null && returnDate != null) {
+            int conversionDifference = 1;
+            int conversionFactor = (1000 * 60 * 60 * 24);
+            int numberOfRentDays = (int) ((returnDate.getTime() - pickUpDate.getTime()) / (conversionFactor)) + conversionDifference;
```

Comment:
You can use https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html